OpenProjects.html

```diff
@@ -447,6 +448,96 @@
   </p>
 </div>
 
+<!-- *********************************************************************** -->
+<div class="www_subsubsection">
+  <a id="llvm_lto_dependency_info">Richer symbol dependency information for LTO</a>
+</div>
+<!-- *********************************************************************** -->
+
+<div class="www_text">
+  <p><b>Description of the project:</b>
+    The implications of LLVM's LTO implementation with respect to linker GC
+    (linker garbage collection) can be improved, especially for aggressive forms
+    with lazy inclusion of objects and sections.
+    In particular, the symbols referenced but undefined by an LTO module are
```

Comment:
I think it would be clearer to describe these as being referenced but undefined by regular native objects included in an LTO link. To me, an "LTO module" is one compiled with bitcode, for which we do have exact reference info during the LTO portion of the link.

Also, I think it would be good to include a little bit of background info about LTO, and about the particular case here where files compiled with LTO (to LLVM bitcode) are LTO linked with files compiled down to native code (regular objects). I would imagine that many GSoC students won't have a lot / any background on LTO.

Reply:
There may be something to say about the symbols referenced but undefined by regular native objects, but being "monolithic at the module level" is not it (especially, from the linker point-of-view). I guess we can say it is "monolithic to LTO".

I did mean "LTO module" in the way you describe though. The linker does not see the granularity of the references.

```suggestion
    In particular, the symbols referenced but undefined by an LTO module are,
    to the linker,
```

Reply:
> I think it would be good to include a little bit of background info about LTO, and about the particular case here where files compiled with LTO (to LLVM bitcode) are LTO linked with files compiled down to native code (regular objects). I would imagine that many GSoC students won't have a lot / any background on LTO.

Great idea. I think there's a submission deadline in about 20 minutes, so I am going to commit what we have first and add more later.

Reply:
It's organization submission deadline, not the project proposal :)

Reply:
> It's organization submission deadline, not the project proposal :)

Thanks for the clarification. @asl, I still think having something posted is safer than not.

Reply:
Absolutely!